easybuild/framework/easyconfig/templates.py

```diff
@@ -205,7 +206,15 @@ def template_constant_dict(config, ignore=None, skip_lower=True):
 
     # step 2: define *ver and *shortver templates
     for name, pref in TEMPLATE_SOFTWARE_VERSIONS:
-        for dep in config.get('dependencies', []):
+
+        # copy to avoid changing original list below
+        deps = copy.copy(config.get('dependencies', []))
+
+        # only consider build dependencies for defining *ver and *shortver templates if we're in iterative mode
+        if hasattr(config, 'iterating') and config.iterating:
```

Comment:
Doesn't it make sense to define iterating to `False` as default, and avoid `hasattr`?

Reply:
The problem is that `config` may be a simple dictionary, rather than an `EasyConfig` instance... And a `dict` value doesn't have a `config` attribute.